AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/MutationSync/OutgoingMutationQueue/ProcessMutationErrorFromCloudOperation.swift

```diff
@@ -33,68 +43,306 @@ class ProcessMutationErrorFromCloudOperation: Operation {
         log.verbose(#function)
 
         guard !isCancelled else {
-            queryOperation?.cancel()
-            let apiError = DataStoreError.unknown("Operation cancelled", "")
-            finish(result: .failure(apiError))
+            let error = DataStoreError.unknown("Operation cancelled", "")
```

Comment:
seems like we need to cancel the mutationOperation ?

Reply:
i think this is fine, the `mutationOperation` is the reference to the API.mutate call, if it's cancelled, the `cancel()` override cancelled would have been called already which calls `mutationOperation.cancel()`, and then here we call `override finish()` which does the clean up and removes the listener